src/OrbitGl/CaptureStatsTest.cpp

```diff
@@ -0,0 +1,103 @@
+  scopes.push_back(CreateScope(/*pid=*/0, /*tid=*/1, /*cpu=*/3, /*start_ns=*/0, /*end_ns=*/10));
+  scopes.push_back(CreateScope(/*pid=*/0, /*tid=*/1, /*cpu=*/4, /*start_ns=*/0, /*end_ns=*/10));
+  scopes.push_back(CreateScope(/*pid=*/0, /*tid=*/1, /*cpu=*/5, /*start_ns=*/0, /*end_ns=*/10));
+  scopes.push_back(CreateScope(/*pid=*/0, /*tid=*/1, /*cpu=*/6, /*start_ns=*/0, /*end_ns=*/10));
+  scopes.push_back(CreateScope(/*pid=*/0, /*tid=*/1, /*cpu=*/7, /*start_ns=*/0, /*end_ns=*/10));
+
+  {
+    SchedulingStats scheduling_stats(scopes, thread_name_provider, /*start_ns=*/0, /*end_ns*/ 1000);
+
+    EXPECT_EQ(scheduling_stats.GetTimeRangeMs(), 0.001);
+    EXPECT_EQ(scheduling_stats.GetTimeOnCoreNs(), 80);
+    EXPECT_EQ(scheduling_stats.GetTimeOnCoreNsByCore().size(), 8);
+    EXPECT_EQ(scheduling_stats.GetProcessStatsByPid().size(), 1);
+    EXPECT_EQ(scheduling_stats.GetProcessStatsSortedByTimeOnCore().size(), 1);
+  }
+
+  scopes.push_back(CreateScope(/*pid=*/1, /*tid=*/2, /*cpu=*/0, /*start_ns=*/0, /*end_ns=*/1000));
+  scopes.push_back(CreateScope(/*pid=*/1, /*tid=*/2, /*cpu=*/1, /*start_ns=*/0, /*end_ns=*/1000));
+  scopes.push_back(CreateScope(/*pid=*/1, /*tid=*/2, /*cpu=*/2, /*start_ns=*/0, /*end_ns=*/1000));
+  scopes.push_back(CreateScope(/*pid=*/1, /*tid=*/2, /*cpu=*/3, /*start_ns=*/0, /*end_ns=*/1000));
+  scopes.push_back(CreateScope(/*pid=*/1, /*tid=*/2, /*cpu=*/4, /*start_ns=*/0, /*end_ns=*/1000));
+  scopes.push_back(CreateScope(/*pid=*/1, /*tid=*/2, /*cpu=*/5, /*start_ns=*/0, /*end_ns=*/1000));
+  scopes.push_back(CreateScope(/*pid=*/1, /*tid=*/2, /*cpu=*/6, /*start_ns=*/0, /*end_ns=*/1000));
+  scopes.push_back(CreateScope(/*pid=*/1, /*tid=*/2, /*cpu=*/7, /*start_ns=*/0, /*end_ns=*/1000));
+
+  scopes.push_back(CreateScope(/*pid=*/2, /*tid=*/3, /*cpu=*/0, /*start_ns=*/0, /*end_ns=*/100));
+  scopes.push_back(CreateScope(/*pid=*/2, /*tid=*/3, /*cpu=*/1, /*start_ns=*/0, /*end_ns=*/100));
+  scopes.push_back(CreateScope(/*pid=*/2, /*tid=*/3, /*cpu=*/2, /*start_ns=*/0, /*end_ns=*/100));
+  scopes.push_back(CreateScope(/*pid=*/2, /*tid=*/3, /*cpu=*/3, /*start_ns=*/0, /*end_ns=*/100));
+  scopes.push_back(CreateScope(/*pid=*/2, /*tid=*/3, /*cpu=*/4, /*start_ns=*/0, /*end_ns=*/100));
+  scopes.push_back(CreateScope(/*pid=*/2, /*tid=*/3, /*cpu=*/5, /*start_ns=*/0, /*end_ns=*/100));
+  scopes.push_back(CreateScope(/*pid=*/2, /*tid=*/3, /*cpu=*/6, /*start_ns=*/0, /*end_ns=*/100));
+  scopes.push_back(CreateScope(/*pid=*/2, /*tid=*/3, /*cpu=*/7, /*start_ns=*/0, /*end_ns=*/100));
+
+  {
+    SchedulingStats scheduling_stats(scopes, thread_name_provider, /*start_ns=*/0, /*end_ns*/ 1000);
+
+    EXPECT_EQ(scheduling_stats.GetTimeRangeMs(), 0.001);
+    EXPECT_EQ(scheduling_stats.GetTimeOnCoreNsByCore().size(), 8);
+    EXPECT_EQ(scheduling_stats.GetProcessStatsByPid().size(), 3);
+    EXPECT_EQ(scheduling_stats.GetProcessStatsSortedByTimeOnCore().size(), 3);
```

Comment:
I would like to also check a little more about detailed information based on the the scopes you put before

Reply:
b/186242694